app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
+      persistent: true,
+      connect_timeout: TIMEOUT)
+  end
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    route = 'spamreports.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def bounced?(email)
-    api { bounces.retrieve(email: email).any? }
+    route = 'bounces.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def remove_from_bounce_list(email)
-    api { bounces.delete(email: email) }
+    route = 'bounces.delete.json'
+    body = { email: email }
+
+    response = post_request(route, body)
+
+    check_email_error(response)
+    true
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
```

Comment:
This code is (unlike `bouced?` it would seem – but it's not obvious from the code) using exceptions for flow control and is using exceptions here for 2 completely separate things:

1. For flow control in the case that `check_email_error` raises an exception (because the email address doesn't exist). This shouldn't be implemented using exceptions in my opinion.
2. In the case of other exceptions – HTTP etc. Again same concern about catching everything.

Also, I think the consumer of this client should be the one to decide that HTTP errors, timeouts, etc. are not important. Returning false in the case of explicit failure (no email) and call failure (API down etc) is not good.